qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -239,16 +294,41 @@ def input_gradients(self, input_gradients: bool) -> None:
         self._input_gradients = input_gradients
         self._construct_gradient_circuit()
 
-    def set_interpret(self, interpret, output_shape=None):
+    def set_interpret(
+        self,
+        interpret: Optional[Callable[[int], Union[int, Tuple[int, ...]]]],
+        output_shape: Union[int, Tuple[int, ...]] = None,
+    ) -> None:
         """Change 'interpret' and corresponding 'output_shape'. If self.sampling==True, the
         output _shape does not have to be set and is inferred from the interpret function.
-        Otherwise, the output_shape needs to be given."""
-        self._interpret = interpret if interpret else lambda x: x
+        Otherwise, the output_shape needs to be given.
+
+        Args:
+            interpret: A callable that maps the measured integer to another unsigned integer or
+                tuple of unsigned integers. See constructor for more details.
+            output_shape: The output shape of the custom interpretation, only used in the case
+                where an interpret function is provided and ``sampling==False``. See constructor
+                for more details.
+
+        Returns:
+            None.
```

Comment:
Not necessary because of the typehint. Please check other places.

Reply:
Removed the return statements.